packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/handlers/InvokeSingleClickOnDiagramElementToolEventHandler.java

```diff
@@ -99,101 +80,34 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
         ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, diagramInput.representationId(), diagramInput);
 
         if (diagramInput instanceof InvokeSingleClickOnDiagramElementToolInput input) {
-            Diagram diagram = diagramContext.getDiagram();
-            var optionalTool = this.toolService.findToolById(editingContext, diagram, input.toolId())
-                    .filter(SingleClickOnDiagramElementTool.class::isInstance)
-                    .map(SingleClickOnDiagramElementTool.class::cast);
-            if (optionalTool.isPresent()) {
-                IStatus status = this.executeTool(editingContext, diagramContext, input.diagramElementId(), optionalTool.get(), input.variables());
-                if (status instanceof Success success) {
-                    WorkbenchSelection newSelection = null;
-                    Object newSelectionParameter = success.getParameters().get(Success.NEW_SELECTION);
-                    if (newSelectionParameter instanceof WorkbenchSelection workbenchSelection) {
-                        newSelection = workbenchSelection;
-                    }
-                    payload = new InvokeSingleClickOnDiagramElementToolSuccessPayload(diagramInput.id(), newSelection, success.getMessages());
-                    changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, diagramInput.representationId(), diagramInput);
-                } else if (status instanceof Failure failure) {
-                    payload = new ErrorPayload(diagramInput.id(), failure.getMessages());
+            IStatus status = this.toolDiagramExecutor.execute(editingContext, diagramContext.getDiagram(), input.toolId(), input.diagramElementId(), input.variables());
```

Comment:
We shouldn't need to modify every part of the execution of any tool in all our representations to start being coupled to this new feature. The dependency should be the other way around. I'm not sure why anything related to the impact analysis should be entangled with the current behavior of Sirius Web.

We shall try to reduce coupling, always. Not add new ones.